### PR TITLE
fix: Build `nyc-test-coverage` package, fixes #839

### LINKED ISSUE #839: [Bug] @temporalio/nyc-test-coverage npm module doesn't contain lib
### What are you really trying to do?

Run `npm install @temporalio/nyc-test-coverage@1.1.0 && ls node_modules/@temporalio/nyc-test-coverage`. You'll see that there's only an `src` directory, no `lib`. Here's a screenshot:

image

### Describe the bug

See above

### Minimal Reproduction

See above

### Environment/Versions

- OS and processor: Linux
- Temporal Version: TypeScript SDK 1.1.0
- Are you using Docker or Kubernetes or building Temporal from source? No

### PR DESCRIPTION
**Pull request by lorensr**
#839

When I added `    "build": "tsc --build",` to nyc package.json, I got this:

image

hence adding to `/test/tsconfig.json` instead.

### CODE REVIEW

File: packages/nyc-test-coverage/package.json [hunk 1]
```diff
@@ -2,16 +2,24 @@
...
+    "url": "https://github.com/temporalio/sdk-typescript/issues"
+  },
   "license": "MIT",
+  "author": "Temporal Technologies Inc. <sdk@temporal.io>",
+  "main": "lib/index.js",
+  "types": "./lib/index.d.ts",
+  "files": [
+    "lib",
+    "src"
+  ],
```
Comment (lorensr) [inline packages/nyc-test-coverage/package.json:22]: this is just `npx sort-package-json`

File: packages/test/tsconfig.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
     { "path": "../interceptors-opentelemetry" },
     { "path": "../testing" },
     { "path": "../worker" },
-    { "path": "../workflow" }
+    { "path": "../workflow" },
+    { "path": "../nyc-test-coverage" }
```
Comment (bergundy) [inline packages/test/tsconfig.json:20]: We should add a test for this too but not blocking this pr

**Review by bergundy — APPROVED** [review]
Thanks @lorensr

**Comment by bennidhamma** [comment]
I'm seeing this issue too. Any sense of when this will fixed in npm?

**Comment by lorensr** [comment]
Early next week

**Comment by bennidhamma** [comment]
I just pulled the files down and built them in the project. Now I'm getting this error (running in jest):

```
      SyntaxError: Unexpected token u in JSON at position 0
          at JSON.parse (<anonymous>)
          at Object.concludeActivation (/home/circleci/project/test/utilities/temp_nyc_temporal/interceptors.ts:16:28)
          at next (/home/circleci/project/node_modules/@temporalio/internal-workflow-common/src/interceptors.ts:28:59)
          at Object.concludeActivation (/home/circleci/project/node_modules/@temporalio/workflow/src/worker-interface.ts:243:23)
          at evalmachine.<anonymous>:1:18
          at Script.runInContext (node:vm:139:12)
          at Object.runInContext (node:vm:289:6)
          at Proxy.<anonymous> (/home/circleci/project/node_modules/@temporalio/worker/lib/workflow/vm.js:107:46)
          at VMWorkflow.activate (/home/circleci/project/node_modules/@temporalio/worker/lib/workflow/vm.js:341:48)
          at async handleRequest (/home/circleci/project/node_modules/@temporalio/worker/lib/workflow/workflow-worker-thread.js:45:32),
      workflowExists: true
    }".

      14 |         const globalCoverage: CoverageMapData = global.__coverage__;
      15 |
    > 16 |         coverage.merge(JSON.parse(JSON.stringify(globalCoverage)));
         |                            ^
      17 |         clearCoverage(globalCoverage);
      18 |
      19 |         return next(input);
```

I'm assuming `global.__coverage__` is `undefined`. I don't know too much about jest or nyc, but it looks like jest manages nyc itself, so maybe no global.__coverage__ then? hmm...

**Pull request merged**